src/WindowTitleBar/CloneButton.tsx

```diff
@@ -0,0 +1,173 @@
+import React from "react";
+import {FinsembleWindow, SpawnParams, WindowBounds} from "@finsemble/finsemble-core";
+
+type CloneButtonProps = {
+
+}
+
+export const CloneButton: React.FunctionComponent<CloneButtonProps> = () => {
+
+	/**
+	 * Clones an app
+	 * @param bounds
+	 * @param addToTabGroup
+	 * @param staggerPosition
+	 */
+	const cloneApp = async (bounds?:WindowBounds, addToTabGroup?:boolean, staggerPosition:number = 25) => {
+		let {component, spawnParams,componentState} = await getComponentData(finsembleWindow);
+		if(bounds) {
+			delete spawnParams.top;
+			delete spawnParams.left;
+			delete spawnParams.width;
+			delete spawnParams.height;
+			spawnParams = {...spawnParams, ...bounds}
+		}
+
+		if(!bounds && staggerPosition) {
+			// Can this always be assumed to be a number?
+			spawnParams.top = spawnParams.top ? spawnParams.top as number + staggerPosition : staggerPosition;
+			spawnParams.left = spawnParams.left ? spawnParams.left as number + staggerPosition : staggerPosition;
+		}
+
+		const {response} = await FSBL.Clients.AppsClient.spawn(component, {...spawnParams, addToWorkspace: true});
+		// Race condition
+		prepareComponentState(spawnParams.name as string,componentState)
```

Comment:
You are right that theres a race here which you'll probably lose sometimes. To resolve some small work is needed in the Launcher and Workspace service.


Workspace service wipes out `componentState` [here](https://github.com/Finsemble/finsemble-mono/blob/main/packages/core/src/platform/services/workspace/workspaceManager.ts/#L1455-L1471) and should take an optional argument to set it instead. The relevant responder is [here](https://github.com/Finsemble/finsemble-mono/blob/main/packages/core/src/platform/services/workspace/workspaceService.ts/#L1095), called from [here](https://github.com/Finsemble/finsemble-mono/blob/ebefe7bb901d6823b65706929d1ace7e7e762ab1/packages/core/src/platform/services/workspace/workspaceAdminAPI.ts#L13) and called from `spawnWithUpdateData` [here](https://github.com/Finsemble/finsemble-mono/blob/ebefe7bb901d6823b65706929d1ace7e7e762ab1/packages/core/src/platform/services/window/Launcher/launcher.ts#L3792).

You'll have to create a new [SpawnParam](https://github.com/Finsemble/finsemble-mono/blob/ebefe7bb901d6823b65706929d1ace7e7e762ab1/packages/core/src/platform/services/window/types.ts#L555) for component state, pass it down through the spawn functions  to here and then pass it onwards to WorkspaceService to set-up.